### PR TITLE
Dreamsofspanking

### PR DESCRIPTION
**Pull request by nocrad349**
Added xpath scraper for dreamsofspanking.com

### CODE REVIEW

File: scrapers/dreamsofspanking.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  sceneScraper:
+    scene:
+      Title:
+        selector: //div[@id="product"]/h3/text()
+      Date:
+        selector: //div[@class="date"]/text()
+        postProcess:
+          - replace:
+            - regex: .*pdated ([0-9]{1,2} [A-z]{3} [0-9]{4})
+              with: $1
```
Comment (bnkai) [inline scrapers/dreamsofspanking.yml:17]: We want the `created ` date not the updated one. Some scenes dont even have the updated one
The below seems to work ok
```yaml
              - regex: Created\s+(\d+\s[a-zA-Z]{3}\s\d{4}).*
                with: $1
```

File: scrapers/dreamsofspanking.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        concat: "\n\n"
+      Performers:
+        Name: //div[@class="performers"][1]/p/a/text()
+      Tags:
+        Name:
+          selector: //p[@class="tags"]/a/text()
+      Studio:
+        Name: 
+          fixed: Dreams of Spanking
+      Image: //video/@poster
```
Comment (bnkai) [inline scrapers/dreamsofspanking.yml:30]:
```suggestion
      Image: //video/@poster|//img[@class="preview"]/@src
```
The above works for some of the scenes that are missing the video

File: scrapers/dreamsofspanking.yml (hunk 1 above)
Comment (nocrad349) [reply scrapers/dreamsofspanking.yml:17]: Some old scenes do miss the updated part, but all the new scenes lack "Created".

File: scrapers/dreamsofspanking.yml (hunk 2 above)
Comment (nocrad349) [reply scrapers/dreamsofspanking.yml:30]: I honestly usually just ignore the "no video" updates. If there is no video there won't be a scene in stash and thus no need to scrape. Or do the scene scrapers also work for galleries?

File: scrapers/dreamsofspanking.yml (hunk 1 above)
Comment (bnkai) [reply scrapers/dreamsofspanking.yml:17]: Are you sure about that ? I checked all the first page and the created date was present.
Did you look at the scene page ? The listing page doesnt matter to us.

**Comment by bnkai** [comment]
I didnt even notice the difference between videos and galleries. The layout is one so with the same selectors and yaml anchors you get a free gallery scraper

```yaml
name: DreamsOfSpanking
sceneByURL:
  - action: scrapeXPath
    url:
      - dreamsofspanking.com/scene/item
    scraper: sceneScraper
galleryByURL:
  - action: scrapeXPath
    url:
      - dreamsofspanking.com/scene/item
    scraper: galleryScraper
xPathScrapers:
  sceneScraper:
    scene:
      Title: &title
        selector: //div[@id="product"]/h3/text()
      Date: &date
        selector: //div[@class="date"]/text()
        postProcess:
          - replace:
              - regex: Created\s+(\d+\s[a-zA-Z]{3}\s\d{4}).*
                with: $1
          - parseDate: 2 Jan 2006
      Details: &details
        selector: //div[@class="long-description"]/p/text()
        concat: "\n\n"
      Performers: &performers
        Name: //div[@class="performers"][1]/p/a/text()
      Tags: &tags
        Name:
          selector: //p[@class="tags"]/a/text()
      Studio: &studio
        Name:
          fixed: Dreams of Spanking
      Image: //video/@poster|//img[@class="preview"]/@src
  galleryScraper:
    gallery:
      Title: *title
      Date: *date
      Details: *details
      Performers: *performers
      Tags: *tags
      Studio: *studio
# Last Updated May 4, 2021
```

File: scrapers/dreamsofspanking.yml (hunk 1 above)
Comment (nocrad349) [reply scrapers/dreamsofspanking.yml]: You're right, I was looking at some single item list pages and didn't see there werent scene pages.
Fixed now

**Comment by nocrad349** [comment]
@bnkai I've added tghe gallery scraper part.

**Pull request merged**